src/GoTrueClient.ts

```diff
@@ -558,6 +562,27 @@ export default class GoTrueClient {
     }
   }
 
+  private async _handleOpenIDConnectSignIn(
+    { IDToken, nonce, clientID, issuer, provider }: OpenIDConnectCredentials
+  ): Promise<{
+    session: Session | null
+    user: User | null
+    error: ApiError | null
+  }> {
+    if (IDToken && nonce && ( (clientID && issuer) || provider) ) {
+      try {
+        const { data, error } = await this.api.signInWithOpenIDConnect({IDToken, nonce, clientID, issuer, provider})
+        if (error || !data) return { user: null, session: null, error }
+        this._saveSession(data)
+        this._notifyAllSubscribers('SIGNED_IN')
+        return { user: data.user, session: data, error: null }
+      } catch (e) {
+        return { user: null, session: null, error: e as ApiError }
+      }
+    }
+    throw new Error(`You must provide a OpenID Connect provider with idtoken and nonce.`)
```

Comment:
```suggestion
    throw new Error(`You must provide a OpenID Connect provider with your id token and nonce.`)
```

Is Client required?

It is optional from the arguments, but is it needed to properly configure OpenID Connect?

Reply:
https://github.com/supabase/gotrue/pull/360

↑The client_id and issuer do not exist until this PR is merged.

Normally, id_token (or idtoken) identifies the oidc, but "id token" does not identify the oidc.
But since it's your request, I'll go with it.